### PR TITLE
Binomial heuristics when n is large and k is small

### PR DESCRIPTION
**Pull request by ethankward**
Currently expressions such as `binomial(gamma(25), 6)` cannot be evaluated since binomial uses a prime sieving method and causes a memory error. If `k` is less than the number of primes less than `n` it is more efficient to simply compute the product from `n-k+1` to `n`. Implemented this and added a test.

### CODE REVIEW

**Comment by asmeurer** [comment]
Can you add a comment to the code to the effect of what you wrote here?

File: sympy/functions/combinatorial/factorials.py [hunk 1]
```diff
@@ -803,27 +806,36 @@ def _eval(self, n, k):
...
+                                a = int((N % prime) < (K % prime + a))
+                                N, K = N // prime, K // prime
+                                exp = a + exp
+
+                            if exp > 0:
+                                result *= prime**exp
+                else:
+                    result = S.One
+                    for i in range(n-k+1, n+1):
+                        result *= i
```
Comment (asmeurer) [inline sympy/functions/combinatorial/factorials.py]: I believe you can use `result = ff(n, k)` here.

**Comment by asmeurer** [comment]
The hyperbolic fix is https://github.com/sympy/sympy/pull/13480/files. It should probably be removed this this PR, unless it is a dependency.

**Comment by ethankward** [comment]
Sorry, I accidentally included the hyperbolic PR. Is there a way for me to remove it?

**Comment by asmeurer** [comment]
Oh I didn't notice they are the exact same commits. In that case, it's not a huge deal, because once both PRs are merged, they will be just the one commit in the git history. In fact, if this PR is merged first, it will automatically close the other one.

**Comment by asmeurer** [comment]
This looks good to me.

**Pull request merged**